### PR TITLE
DYN-1545: Build performance test suite

### PR DESCRIPTION
**Pull request by scottmitchell**
### Purpose

[DYN-1545: Build performance test suite](https://jira.autodesk.com/browse/DYN-1545)
This PR adds an assortment of relatively "heavy" Dynamo graphs to be used with the Dynamo performance benchmarking tool. Together, these graphs achieve coverage of a variety of Dynamo use cases:

graphs

> SinuousTower removed because it takes over a minute to run and its coverage is redundant.
> LotsOfStuff removed because it is covered by LotsOfColoredStuff.

I've also added some info about the graphs to the [DynamoPerformanceTests readme](https://github.com/scottmitchell/Dynamo/tree/DYN-1545/tools/Performance/DynamoPerformanceTests), including a screenshot of each graph—to help understand what each is testing. I'm looking for feedback about the value of including each of these tests, whether we want more tests, fewer, etc. Most of the tests that I'm adding here are large and cover a number of different Dynamo use cases. It may make sense to include these large graphs, and to also add smaller, more targeted graphs. 

This set of graphs takes 38 minutes to run locally on my machine. Here are the current benchmark results for these graphs:

performance

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### FYIs

@mjkkirschner @aparajit-pratap @QilongTang @alfarok @reddyashish @ColinDayOrg

### CODE REVIEW

**Comment by QilongTang** [comment]
Maybe worth to alter the version number to older and make sure they can be opened starting from Dynamo 2.0.2?

File: tools/Performance/DynamoPerformanceTests/graphs/12-IsovistTest_OOTB.dyn [hunk 1]
```diff
@@ -0,0 +1,9971 @@
...
+          "UsingDefaultValue": true,
+          "Level": 2,
+          "UseLevels": false,
+          "KeepListStructure": false
+        },
+        {
+          "Id": "1c16497a6fac4d3cb52582dd29a2265d",
+          "Name": "highPoint",
+          "Description": "Point\nDefault value : Autodesk.DesignScript.Geometry.Point.ByCoordinates(1, 1, 1)",
+
```
Comment (mjkkirschner) [inline tools/Performance/DynamoPerformanceTests/graphs/12-IsovistTest_OOTB.dyn]: @scottmitchell I think this is what @QilongTang meant - sometimes opening graphs from new versions of Dynamo in older versions will pop up a blocking message dialog ("this is from the future") and that may throw off benchmarks if it stops the graphs from running.

Comment (scottmitchell) [reply tools/Performance/DynamoPerformanceTests/graphs/12-IsovistTest_OOTB.dyn]: @mjkkirschner @QilongTang I saved each graph out from 2.0.2, except for `PythonJSONParsing.dyn` which uses JSON nodes and requires 2.1.0

Comment (QilongTang) [reply tools/Performance/DynamoPerformanceTests/graphs/12-IsovistTest_OOTB.dyn]: Excellent!

**Comment by QilongTang** [comment]
LGTM

**Comment by mjkkirschner** [comment]
If it requires 2.1 features then running it in 2.0.2 will likely not work so great :) We'll need to exclude it I guess from past comparisons.

**Comment by QilongTang** [comment]
@mjkkirschner @scottmitchell We can do code changes later to enable exclusion of any graphs in particular Dynamo Version.

**Comment by scottmitchell** [comment]
@mjkkirschner @QilongTang I added info about Dynamo version compatibility for each graph to the [readme](https://github.com/scottmitchell/Dynamo/blob/DYN-1545/tools/Performance/DynamoPerformanceTests/README.md) so we can keep track of that.

**Pull request merged**